lib/vtls/openssl.c

```diff
@@ -2720,6 +2726,157 @@ static CURLcode ossl_connect_step1(struct connectdata *conn, int sockindex)
   }
 #endif
 
+
+#if defined(USE_WIN32_CRYPTO) && \
+    !defined(LIBCURL_WINDOWS_OPENSSL_NEVER_USE_WINDOWS_CA_STORE)
+  /* Import certificates from the Windows root certificate store if requested.
+     https://stackoverflow.com/questions/9507184/
+     https://github.com/d3x0r/SACK/blob/master/src/netlib/ssl_layer.c#L1037
+     https://tools.ietf.org/html/rfc5280 */
+  if((SSL_CONN_CONFIG(verifypeer) || SSL_CONN_CONFIG(verifyhost)) &&
+     !ssl_capath && ssl_cafile && !strcmp(ssl_cafile, CURL_WINDOWS_CA_STORE)) {
+    X509_STORE *store = SSL_CTX_get_cert_store(BACKEND->ctx);
+    HCERTSTORE hStore = CertOpenSystemStoreA((HCRYPTPROV_LEGACY)NULL, "ROOT");
+
+    if(hStore) {
+      PCCERT_CONTEXT pContext = NULL;
+      /* The array of enhanced key usage OIDs will vary per certificate and is
+         declared outside of the loop so that rather than malloc/free each
+         iteration we can grow it with realloc, when necessary. */
+      CERT_ENHKEY_USAGE *enhkey_usage = NULL;
+      DWORD enhkey_usage_size = 0;
+
+      /* This loop makes a best effort to import all valid certificates from
+         the MS root store. If a certificate cannot be imported it is skipped.
+         'result' is used to store only hard-fail conditions (such as out of
+         memory) that cause an early break. */
+      result = CURLE_OK;
+      for(;;) {
+        X509 *x509;
+        FILETIME now;
+        BYTE key_usage[2];
+        DWORD req_size;
+        const unsigned char *encoded_cert;
+#ifdef CURL_VERBOSE_OSSL_MS_IMPORT
```

Comment:
Who sets this define? Shouldn't this rather use the plain old `#ifdef DEBUGBUILD` ?

Reply:
You are right, I replace with
#if defined(DEBUGBUILD) && !defined(CURL_DISABLE_VERBOSE_STRINGS)